### PR TITLE
*: Remove tidb-binlog related code

### PR DESCRIPTION
**Pull request by Benjamin2037**
Issue Number: ref #55949

Problem Summary:
Update tipb and remove binlog code.

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [X] No need to test
  > - [ ] I checked and no code files have been changed.
  > 
  Only delete code.

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: pkg/testkit/testkit.go [hunk 1]
```diff
@@ -686,16 +684,6 @@ func (c *RegionProperityClient) SendRequest(ctx context.Context, addr string, re
 // MockPumpClient is a mock pump client.
 type MockPumpClient struct{}
```
Comment (lcwangchao) [inline pkg/testkit/testkit.go:685]: MockPumpClient should also be deleted?

File: tests/realtikvtest/brietest/binlog_test.go [hunk 2]
```diff
@@ -15,39 +15,17 @@
...
 	"github.com/pingcap/tidb/pkg/parser/mysql"
-	"github.com/pingcap/tidb/pkg/store/mockstore/mockcopr"
 	"github.com/pingcap/tidb/pkg/testkit"
 	"github.com/pingcap/tidb/tests/realtikvtest"
-	"github.com/pingcap/tipb/go-binlog"
 	"github.com/stretchr/testify/require"
-	"google.golang.org/grpc"
 )
 
 type mockBinlogPump struct {
```
Comment (lcwangchao) [inline tests/realtikvtest/brietest/binlog_test.go:26]: We can delete `mockBinlogPump` too.

**Comment by lcwangchao** [comment]
Some defs in parser can also be removed:

https://github.com/pingcap/tidb/blob/cf954e2057694a7fc3539a45aecff1fadc4bd332/pkg/parser/ast/misc.go#L76-L77

**Review by tangenta — APPROVED** [review]

**Pull request merged**